### ISSUE TITLE
issues.new: true causes revgrep failure on git 2.39.3 (Apple Git-145)

### ISSUE THREAD

**Issue by PiotrZakrzewski-instruqt** [golangci/golangci-lint#4180]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Trying to use `issues.new: true` setting results in

```
WARN [runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff: exit status 255 
```
while this exact same repo and golangci-lint version (v1.55.0) works fine when using homebrew installed git (git version 2.41.0)

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.55.0 built with go1.21.3 from (unknown, mod sum: "h1:ePpc6YhM1ZV8kHU8dwmHDHAdeedZHdK8cmTXlkkRdi8=") on (unknown)
```

</details>

### Configuration

<details>

```console
issues:
  new: true
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.21.3 darwin/arm64
GO111MODULE=''
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/piotrzakrzewski/Library/Caches/go-build'
GOENV='/Users/piotrzakrzewski/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/piotrzakrzewski/go/pkg/mod'
GONOPROXY='go.instruqt.com/gozilla'
GONOSUMDB='go.instruqt.com/gozilla'
GOOS='darwin'
GOPATH='/Users/piotrzakrzewski/go'
GOPRIVATE='go.instruqt.com/gozilla'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.21.3'
GCCGO='gccgo'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='1'
GOMOD='/Users/piotrzakrzewski/work/gozilla2/go.mod'
GOWORK='/Users/piotrzakrzewski/work/gozilla2/go.work'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/52/n8rhgnzd121944nnlbwhyl8w0000gp/T/go-build4067647609=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Used config file .golangci-lint.yml 
INFO [lintersdb] Active 13 linters: [bodyclose errcheck forbidigo goimports gosec gosimple govet ineffassign nilerr nilnil sqlclosecheck staticcheck unused] 
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|name|deps|files|imports|types_sizes) took 11.289675959s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 35.519125ms 
INFO [linters_context/goanalysis] analyzers took 6m35.14691932s with top 10 stages: buildir: 2m10.207400392s, buildssa: 47.390712581s, fact_deprecated: 8.320884843s, goimports: 6.897477094s, inspect: 6.471722243s, printf: 5.842608809s, gosec: 5.308449454s, ctrlflow: 4.766217217s, nilness: 4.374443119s, forbidigo: 4.359614047s 
WARN [runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff: exit status 255 
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// no GO code needed
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
Duplicate of #4152, #4159

Fixed in v1.55.1
